services/src/test/java/org/exoplatform/addons/gamification/storage/RuleStorageTest.java

```diff
@@ -225,4 +227,34 @@ public void testDeleteRule() {
       fail("Error to add rule", e);
     }
   }
+
+  @Test
+  public void testFindAllChallengesByUserByDomain() {
+    try {
+      assertEquals(ruleStorage.findAllChallengesByUserByDomain(0, 0, 0, Collections.singletonList(0l)).size(), 0);
+      DomainEntity domain1 = newDomain("domain1");
+      DomainEntity domain2 = newDomain("domain2");
+      newChallenge("rule1", domain1.getTitle());
+      newChallenge("rule2", domain1.getTitle());
+      newChallenge("rule3", domain2.getTitle());
+      assertEquals(ruleStorage.findAllChallengesByUserByDomain(domain1.getId(), 0, 4, Collections.singletonList(1l)).size(), 2);
+    } catch (Exception e) {
```

Comment:
it's better to not introduce a try/catch, even if this is a minor issue because you already introduced a `fail` command